### DISCUSSION TITLE
Performance. Table rerender, when typing in Column TextField.

### DISCUSSION THREAD

**Discussion by dukewf04** [KevinVandy/material-react-table#1203] [Q&A] [answered]
Hello everyone. When i typing in textfield (in columns) i see a little lag, full table rerender once. How can avoid rerendering full table?

**Comment by KevinVandy** [comment, marked as answer]
Debouncing filter values is built in under the hood to prevent the total number of renders.

When any of the state charges, the whole table will rerender. Nothing will change that.

**Comment by dukewf04** [comment]
@KevinVandy thank for answer! But can i do something, when i type in TextFilter, the table has lag for 0.5 seconds about (and i can do nothing)... Table has 100 rows and 8 columns.

**Reply by KevinVandy** [reply]
You can hook directly into the onchange event for the filter textfield. That's immediate. The table state will just always be debounced for performance reasons.
